### PR TITLE
Superpmi on Microbenchmarks

### PR DESCRIPTION
**Pull request by kunalspathak**
Perform automated superpmi collection on Microbenchmarks during their execution.

- Windows x64/x86/arm64
- Linux x64

Currently, I have disable the collection for Linux arm/arm64 because it needs special handling to perform "dotnet MicroBenchmarks.dll" inside a `$TMP` folder because during execution of benchmarks, we also perform build of boiler plate code which is not allowed in docker since the directory is readonly.

Contributes to https://github.com/dotnet/runtime/issues/43227

### CODE REVIEW

**Comment by kunalspathak** [comment]
Run: https://dev.azure.com/dnceng/internal/_build/results?buildId=992885&view=results

**Comment by kunalspathak** [comment]
@dotnet/jit-contrib

File: src/coreclr/scripts/superpmi_benchmarks.py [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+import argparse
+import re
+import sys
+
+import stat
+from os import path
+from os.path import isfile
+from shutil import copyfile
+from coreclr_arguments import *
+from superpmi import ChangeDir, TempDir
```
Comment (BruceForstall) [inline src/coreclr/scripts/superpmi_benchmarks.py:23]: I'm not fond of anyone depending on superpmi.py; it makes it more fragile to change without potentially breaking (unknown) dependents.

**Review by BruceForstall — APPROVED** [review]
LGTM

File: src/coreclr/scripts/superpmi_benchmarks.py (hunk 1 above)
Comment (kunalspathak) [reply src/coreclr/scripts/superpmi_benchmarks.py:23]: I agree. There are few methods in `super_setup.py` as well that I will extract in a `superpmi_utils.py` file in a future PR.

**Pull request merged**